### PR TITLE
Track submitted jobs

### LINKED ISSUE #748: Measure active connections
## Tell us about the problem you're trying to solve
We need a way to distinguish active connections from inactive ones metrics-wise.
Why? So that we know when a company is actually using us for their data pipelines, and we can build case studies with them (and eventually upgrade them). 
The number of deployments is not a good KPI. 

## Describe the solution you’d like
First, we should agree on what "active" connection means. 1st suggestion: it is a connection with a recurring frequency through which data flows every day.

Then we could send a Segment event on the backend side of the app, every time a sync is done. Having this info on Segment will help us leverage it in any tool we want. 
I added a suggestion of event, named "Connection Sync" here: https://docs.google.com/spreadsheets/d/1qQPMBdz86yMBiyXpjw91D3HoP6hN7yIPRmLrF7xWOgA/edit?usp=sharing

## Additional context
I don't think we log syncs in any way on our side. I'm not sure this is a good solution though.
For the moment, with Segment, we only track when a user sets up a connection, but nothing about active connections.

@cgardens

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Closes https://github.com/airbytehq/airbyte/issues/748

### CODE REVIEW

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigHelpers.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.StandardSync;
+import java.util.UUID;
+
+public class ConfigHelpers {
+
+  public static StandardSourceDefinition getSourceDefinitionFromSource(ConfigRepository configRepository, UUID sourceId) {
```
Comment (michel-tricot) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigHelpers.java:36]: Why didn't you put these in the ConfigRepository Interface?

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigHelpers.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    final SourceConnection source;
```
Comment (michel-tricot) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigHelpers.java:37]: decl + assignement

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 3]
```diff
@@ -67,6 +81,7 @@ public void run() {
   }
 
   private void submitJob(Job job) {
+
```
Comment (michel-tricot) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:84]: NIT: remove line

**Review by michel-tricot — APPROVED** [review]

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 4]
```diff
@@ -89,6 +104,46 @@ private void submitJob(Job job) {
...
+          // no op because this will be noisy as heck.
+        }
+        case SYNC -> {
+          final StandardSourceDefinition sourceDefinition = ConfigHelpers
+              .getSourceDefinitionFromConnection(configRepository, UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+          final StandardDestinationDefinition destinationDefinition = ConfigHelpers
+              .getDestinationDefinitionFromConnection(configRepository, UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+
+          metadata.put("source_name", sourceDefinition.getName());
+          metadata.put("source_id", sourceDefinition.getSourceDefinitionId());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:135]:
```suggestion
          metadata.put("source_definition_id", sourceDefinition.getSourceDefinitionId());
```

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 5]
```diff
@@ -89,6 +104,46 @@ private void submitJob(Job job) {
...
+        case GET_SPEC -> {
+          // no op because this will be noisy as heck.
+        }
+        case SYNC -> {
+          final StandardSourceDefinition sourceDefinition = ConfigHelpers
+              .getSourceDefinitionFromConnection(configRepository, UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+          final StandardDestinationDefinition destinationDefinition = ConfigHelpers
+              .getDestinationDefinitionFromConnection(configRepository, UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+
+          metadata.put("source_name", sourceDefinition.getName());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:134]:
```suggestion
          metadata.put("source_definition_name", sourceDefinition.getName());
```

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 6]
```diff
@@ -89,6 +104,46 @@ private void submitJob(Job job) {
...
+          metadata.put("source_id", sourceDefinition.getSourceDefinitionId());
+          metadata.put("destination_name", destinationDefinition.getName());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:136]:
```suggestion
          metadata.put("destination_definition_name", destinationDefinition.getName());
```

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 7]
```diff
@@ -89,6 +104,46 @@ private void submitJob(Job job) {
...
+          metadata.put("destination_id", destinationDefinition.getDestinationDefinitionId());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:137]:
```suggestion
          metadata.put("destination_definition_id", destinationDefinition.getDestinationDefinitionId());
```

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 8]
```diff
@@ -89,6 +104,46 @@ private void submitJob(Job job) {
...
 
+  private void track(Job job, ConfigRepository configRepository) {
+    try {
+      final Builder<String, Object> metadata = ImmutableMap.builder();
+      switch (job.getConfig().getConfigType()) {
+        case CHECK_CONNECTION_SOURCE, DISCOVER_SCHEMA -> {
+          final StandardSourceDefinition sourceDefinition = ConfigHelpers
+              .getSourceDefinitionFromSource(configRepository, UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+
+          metadata.put("source_name", sourceDefinition.getName());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:115]:
```suggestion
          metadata.put("source_definition_name", sourceDefinition.getName());
```

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 9]
```diff
@@ -89,6 +104,46 @@ private void submitJob(Job job) {
...
+          metadata.put("source_id", sourceDefinition.getSourceDefinitionId());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:116]:
```suggestion
          metadata.put("source_definition_id", sourceDefinition.getSourceDefinitionId());
```

**Review by sherifnada — APPROVED** [review]

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigHelpers.java (hunk 1 above)
Comment (cgardens) [reply airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigHelpers.java:36]: The exception handling is different. ConfigRepository rethrows everything. My goal here is to swallow exceptions. Wasn't sure if that messed up the contract you had in mind for ConfigRepository.

**Pull request merged**